include/dynd/kernels/arithmetic.hpp

```diff
@@ -430,41 +486,69 @@ namespace nd {
 
 namespace ndt {
 
-  template <type_id_t Src0TypeID>
-  struct type::equivalent<nd::plus_kernel<Src0TypeID>> {
-    typedef typename dynd::type_of<Src0TypeID>::type A0;
-    typedef decltype(+std::declval<A0>()) R;
-
-    static type make()
-    {
-      std::map<std::string, ndt::type> tp_vars;
-      tp_vars["A0"] = ndt::type::make<A0>();
-      tp_vars["R"] = ndt::type::make<R>();
-
-      return ndt::substitute(ndt::type("(A0) -> R"), tp_vars, true);
-    }
-  };
-
-  template <type_id_t Src0TypeID>
-  struct type::equivalent<nd::minus_kernel<Src0TypeID>> {
-    typedef typename dynd::type_of<Src0TypeID>::type A0;
-    typedef decltype(-std::declval<A0>()) R;
-
-    static type make()
-    {
-      std::map<std::string, ndt::type> tp_vars;
-      tp_vars["A0"] = ndt::type::make<A0>();
-      tp_vars["R"] = ndt::type::make<R>();
-
-      return ndt::substitute(ndt::type("(A0) -> R"), tp_vars, true);
-    }
-  };
+#define DYND_DEF_UNARY_OP_KERNEL_EQUIVALENT(OP, NAME)                  \
+  template <type_id_t Src0TypeID>                                      \
+  struct type::equivalent<nd::NAME ## _kernel<Src0TypeID>> {           \
+    typedef typename dynd::type_of<Src0TypeID>::type A0;               \
+    typedef decltype(OP std::declval<A0>()) R;                         \
+                                                                       \
+    static type make()                                                 \
+    {                                                                  \
+      std::map<std::string, ndt::type> tp_vars;                        \
+      tp_vars["A0"] = ndt::type::make<A0>();                           \
+      tp_vars["R"] = ndt::type::make<R>();                             \
+                                                                       \
+      return ndt::substitute(ndt::type("(A0) -> R"), tp_vars, true);   \
+    }                                                                  \
+  };                                                                   \
+
```

Comment:
I'm the one who originally write the make function here, so I'm just criticizing myself. We shouldn't use ndt::substitute in this functions. At the time, we lacked tools to easily construct a callable_type, but we have those now. One should use something like `callable_type::make(ndt::type::make<R>(), {ndt::type::make<A0>()})`.